lib/core/uploaderError.js

```diff
@@ -17,29 +17,43 @@
 
 var sundial = require('sundial');
 
-// Create a new object, that prototypally inherits from the Error constructor.
-function UploaderError(message, stage, error) {
-
-  this.name = this.constructor.name;
-  this.message = message;
+// if another error is passed in then attach it
```

Comment:
So one of our goals is to have good info so we can track down and solve issues. I would rather have more than less at this stage but if over time it is always redundant then we can axe it or if there is a better way to expose it then we can adapt at that time. 

I can separately look at removing the \ as I agree that makes it hard to read.